src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3860,10 +3849,6 @@ export class StudyViewPageStore
                             this._clinicalDataFilterSet.has(uniqueKey) ||
                             this.isInitialFilterState
                         ) {
-                            // return empty if there are no sample identifiers or study ids in the filter
-                            if (!this.hasFilteredSamples) {
```

Comment:
@alisman this is an example of what we were talking about - hasFilteredSamples references selectedSamples, so selectedSamples was being awaited in order to make sure hasFilteredSamples was ready. too opaque

Reply:
but why did we want to short circuit in case of not having filtered samples?  was that an optimization?

Reply:
i realize that you didn't create that code but maybe we should understand exactly why that was

Reply:
as far as I understand, it is just to not make unnecessary calls to the server